### PR TITLE
8.0 fix old style computed fields dle

### PR DESCRIPTION
**Pull request by gurneyalex**
This is a port of https://github.com/odoo/odoo/pull/6315 to OCB with my additional (yet unmerged) PR https://github.com/odoo-dev/odoo/pull/106 included (adding an automated test). 

It fixes the following bug: when records are created through a workflow chain, some stored function fields may not be computed at record creation. This happens for stock.move in the prepaid sale order flow. 

See #6053 for the discussion.

### CODE REVIEW

**Comment by lmignon** [comment]
@gurneyalex Thank you for the tests!!! At least for this bug we are now sure that no regression will occur in the future.
I'm a little surprised to see that part of the fix is to go back to the old API ...

**Comment by gurneyalex** [comment]
@lmignon the "going back to new API" is a revert of an ad-hoc fix (see commit message) which only fixed the stock.move product_qty field, and it is not clear to anyone why changing that field to the new API solved anything at that time. It was committed in 8.0 in some rush because not having product_qty on stock_move completely blocked the flow, but the bug potentially affects many other field...

**Comment by gurneyalex** [comment]
A good way of testing if your instances are affected by this bug is to go through the stored function fields in the database and look for unexepected NULL values. You should have none of these for numeric fields or character fields (they get a value of 0 or '' respectively). A NULL in there means that the compute function was not called at all.

**Comment by lmignon** [comment]
@gurneyalex The first commit should never have happened. Once again thank you for your tests.

File: addons/stock/stock.py [hunk 1]
```diff
@@ -1591,7 +1590,7 @@ def action_traceability(self, cr, uid, ids, context=None):
 # Move
 # ----------------------------------------------------
 
-class stock_move(models.Model):
+class stock_move(osv.osv):
```
Comment (pedrobaeza) [inline addons/stock/stock.py:1593]: Why replace this?

Comment (gurneyalex) [inline addons/stock/stock.py:1593]: see my answer to lmignon's question

Comment (pedrobaeza) [inline addons/stock/stock.py:1593]: Ah, OK, curious that this was introduce upstream..

**Comment by pedrobaeza** [comment]
Seems OK for me :+1:

**Comment by guewen** [comment]
:+1:

**Comment by lmignon** [comment]
:+1:

**Comment by gurneyalex** [comment]
upstream PR updated -> I'll update this one.

**Comment by gurneyalex** [comment]
Upstream PR merged. Closing.

**Pull request closed without merging**